src/test/java/org/example/FilteringSearching.java

```diff
@@ -1,5 +1,147 @@
+    private final By sortByDropDown = By.xpath("//span[@data-csa-c-func-deps='aui-da-a-dropdown-button']");
+    private final By lowToHigh = By.id("s-result-sort-select_1");
+    private final By highToLow = By.id("s-result-sort-select_2");
+
+
+    @BeforeClass
+    private void setDriver() {
+        System.setProperty("webdriver.chrome.driver", "src/test/resources/webdriver/chromedriver.exe");
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+    }
+
+    @AfterClass
+    private void closeDriver() {
+        driver.quit();
+    }
+
+    @Test
+    public void testSearchingBrandedItems() {
+        openPage(AMAZON_URL);
+        clickRandomElement(categories);
+        clickRandomElement(featuredBrands);
+        Assert.assertTrue(verifyMostResultsContainBrandName());
+    }
+
+    @Test
+    private void testSearchingWithinPriceRange() {
+        openPage(AMAZON_URL);
+        clickRandomElement(categories);
+        clickRandomElement(featuredBrands);
+        int minValue = new Random().nextInt(10);
+        int maxValue = new Random().nextInt(minValue, 200);
+        filterResultsByPriceRange(minValue, maxValue);
+        Assert.assertTrue(verifyMostPricesAreWithinRange(minValue, maxValue));
+    }
+
+    @Test
+    private void testSortingResultsByPrice() {
+        openPage(AMAZON_URL);
+        clickRandomElement(categories);
+        clickRandomElement(featuredBrands);
+        click(sortByDropDown);
+        click(lowToHigh);
+        Assert.assertTrue(verifyIfCorrectlySorted(Comparator.naturalOrder()));
+        click(sortByDropDown);
+        click(highToLow);
+        Assert.assertTrue(verifyIfCorrectlySorted(Comparator.reverseOrder()));
+    }
+
+    private boolean verifyMostResultsContainBrandName() {
+        String brandName = getElement(checkedCheckbox).getText().toLowerCase();
+        List<WebElement> results = getElements(searchResults);
+        long numberOfResultsThatContainBrandName = results.stream()
+                .map(WebElement::getText)
+                .map(String::toLowerCase)
+                .filter(elem -> elem.contains(brandName))
+                .count();
+        return numberOfResultsThatContainBrandName >= results.size() / 2;
+    }
+
+    private boolean verifyMostPricesAreWithinRange(int minValue, int maxValue) {
+        List<WebElement> results = getElements(priceResults);
+        long numberOfPricesWithinRange = results.stream()
+                .map(elem -> elem.getAttribute("textContent"))
+                .map(price -> Double.valueOf(price.substring(1)))
+                .filter(price -> price >= minValue && price <= maxValue)
+                .count();
+        return numberOfPricesWithinRange >= results.size() / 2;
+    }
+
+    private boolean verifyIfCorrectlySorted(Comparator<Double> comparator) {
+        List<Double> priceResults = getElements(this.priceResults).stream()
```

Comment:
You can remove `this` statement since you are not instantiating an object based on this class and leave just priceResults

Reply:
Fixed.